### PR TITLE
e2e tests: use orchestrator-plugin v2 api

### PR DESCRIPTION
**Pull request by gciavarrini**
v1 endpoints have been removed, so need to update the tests to use v2 endpoints.

### CODE REVIEW

File: e2e/move2kube.sh
Comment (gabriel-farache) [inline e2e/move2kube.sh:1]: I think you also need to update the calls to /execute no?

Comment (gciavarrini) [reply e2e/move2kube.sh:1]: I missed that one, added. 
Thanks for the catch

**Comment by gabriel-farache** [comment]
@gciavarrini it seems that the API v2 also changed the body format for the `execute`, we need to add `inputData` as root key, ie:
```
out=$(curl -XPOST -H "Content-Type: application/json" -H "Authorization: Bearer ${BACKEND_SECRET}" \
    ${BACKSTAGE_URL}/api/orchestrator/workflows/m2k/execute \
    -d "{\"inputData\": {\"repositoryURL\": \"ssh://${GIT_REPO}\", \"recipients\": [\"user:default/guest\"], \"sourceBranch\": \"${GIT_SOURCE_BRANCH}\", \"targetBranch\": \"${GIT_TARGET_BRANCH}\", \"workspaceId\": \"${WORKSPACE_ID}\", \"projectId\": \"${PROJECT_ID}\"}}")
```
Same for MTAs

File: e2e/move2kube.sh [hunk 1]
```diff
@@ -53,16 +53,16 @@ WORKSPACE_ID=$(curl -X POST "${MOVE2KUBE_URL}/api/v1/workspaces" -H 'Content-Typ
...
 sleep 5
-M2K_STATUS=$(curl -XGET -s -o /dev/null -w "%{http_code}" -H "Authorization: Bearer ${BACKEND_SECRET}" ${BACKSTAGE_URL}/api/orchestrator/workflows/m2k)
+M2K_STATUS=$(curl -XGET -s -o /dev/null -w "%{http_code}" -H "Authorization: Bearer ${BACKEND_SECRET}" ${BACKSTAGE_URL}/api/orchestrator/v2/workflows/m2k/overview)
 done
 
 echo "M2K is available in backstage, sending execution request"
 out=$(curl -XPOST -H "Content-Type: application/json" -H "Authorization: Bearer ${BACKEND_SECRET}" \
-    ${BACKSTAGE_URL}/api/orchestrator/workflows/m2k/execute \
+    ${BACKSTAGE_URL}/api/orchestrator/v2/workflows/m2k/execute \
     -d "{\"repositoryURL\": \"ssh://${GIT_REPO}\", \"recipients\": [\"user:default/guest\"], \"sourceBranch\": \"${GIT_SOURCE_BRANCH}\", \"targetBranch\": \"${GIT_TARGET_BRANCH}\", \"workspaceId\": \"${WORKSPACE_ID}\", \"projectId\": \"${PROJECT_ID}\"}")
```
Comment (gabriel-farache) [inline e2e/move2kube.sh:66]: Should now be
```
-d "{\"inputData\": {\"repositoryURL\": \"ssh://${GIT_REPO}\", \"recipients\": [\"user:default/guest\"], \"sourceBranch\": \"${GIT_SOURCE_BRANCH}\", \"targetBranch\": \"${GIT_TARGET_BRANCH}\", \"workspaceId\": \"${WORKSPACE_ID}\", \"projectId\": \"${PROJECT_ID}\"}}")
```

File: e2e/mta-v6.x.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ echo "Proxy Janus-idp port ✅"
 echo "End to end tests start ⏳"
 
 out=$(curl -XPOST -H "Content-Type: application/json" -H "Authorization: Bearer ${BACKEND_SECRET}" \
-    http://localhost:9080/api/orchestrator/workflows/mta-analysis-v6/execute -d '{"repositoryURL": "https://github.com/spring-projects/spring-petclinic", "recipients": ["user:default/guest"], "exportToIssueManager": "false", "migrationStartDatetime" : "2024-07-01T00:00:00Z", "migrationEndDatetime" : "2024-07-31T00:00:00Z"}')
+    http://localhost:9080/api/orchestrator/v2/workflows/mta-analysis-v6/execute -d '{"repositoryURL": "https://github.com/spring-projects/spring-petclinic", "recipients": ["user:default/guest"], "exportToIssueManager": "false", "migrationStartDatetime" : "2024-07-01T00:00:00Z", "migrationEndDatetime" : "2024-07-31T00:00:00Z"}')
```
Comment (gabriel-farache) [inline e2e/mta-v6.x.sh:33]: Should now be
```
-d '{"inputData": {"repositoryURL": "https://github.com/spring-projects/spring-petclinic", "recipients": ["user:default/guest"], "exportToIssueManager": "false", "migrationStartDatetime" : "2024-07-01T00:00:00Z", "migrationEndDatetime" : "2024-07-31T00:00:00Z"}}'
```

File: e2e/mta.sh [hunk 3]
```diff
@@ -30,7 +30,7 @@ echo "Proxy Janus-idp port ✅"
 echo "End to end tests start ⏳"
 
 out=$(curl -XPOST -H "Content-Type: application/json"  -H "Authorization: Bearer ${BACKEND_SECRET}" \
-    http://localhost:9080/api/orchestrator/workflows/MTAAnalysis/execute \
+    http://localhost:9080/api/orchestrator/v2/workflows/MTAAnalysis/execute \
     -d '{"repositoryURL": "https://github.com/spring-projects/spring-petclinic", "recipients": ["user:default/guest"]}')
```
Comment (gabriel-farache) [inline e2e/mta.sh:34]: Should now be 
```
-d '{"inputData": {"repositoryURL": "https://github.com/spring-projects/spring-petclinic", "recipients": ["user:default/guest"]}}')
```

**Review by gabriel-farache — APPROVED** [review]

**Pull request merged**